### PR TITLE
Lesson2

### PR DESCRIPTION
**Pull request by artpetrovskii**

### CODE REVIEW

File: Blender.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class Blender
+  def close_lid
+    puts "Sealed tight!"
+  end
+  def blend (speed)
```
Comment (apetrovskiy) [inline Blender.rb]: There should be an empty line between methods.

File: animals.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    puts "#{name} runs to the #{destination}."
+  end
+end
+
+bird = Bird.new
+dog = Dog.new
+cat = Cat.new
+
+bird.move("tree")
+dog.talk
```
Comment (apetrovskiy) [inline animals.rb]: The talk method requires one argument

File: animals.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+bird.talk
+cat.move("house")
```
Comment (apetrovskiy) [inline animals.rb]: The move method requires two arguments.

File: Blender.rb (hunk 1 above)
Comment (apetrovskiy) [inline Blender.rb]: rubocop: there should be an empty line,

File: dog.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
+class Dog
+
+  attr_reader :name, :age
+
+  def name=(value)
+    if value == ""
+      raise "Name can't be blank!"
+    end
+  @name = value
```
Comment (apetrovskiy) [inline dog.rb]: indentation

File: dog.rb [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  end
+
+  def age=(value)
+    if value < 0
+      raise "An age of #{value} isn't valid!"
+    end
+  @age = value
```
Comment (apetrovskiy) [inline dog.rb]: indentation

File: mage.rb [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+  end
+
+end
+
+merlin = Mage.new
+merlin.name = "Merlin"
+morgana = Mage.new
+morgana.name = "Morgana"
+morgana.spell = "Shrink"
+morgana.enchant(merlin)
```
Comment (apetrovskiy) [inline mage.rb]: There're numerous style issues
image
image

File: MyClass.rb [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+  end
+  def my_attribute
+    @my_attribute
+  end
+end
+
+my_instance = MyClass.new
+my_instance.my_attribute = "assigned via method call"
+puts my_instance.my_attribute
+my_instance.my_attribute=("same here")
```
Comment (apetrovskiy) [inline MyClass.rb]: There're extra parentheses while lack of spaces

File: Widget.rb [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+  def set_size
+    @size = 'large'
+  end
+  def show_size
+    puts "Size: #{@size}"
+  end
+end
+
+widget = Widget.new
+widget.show_size
```
Comment (apetrovskiy) [inline Widget.rb]: The script doesn't work:
```
[Running] ruby "c:\Projects\Artemii\Head-First-Ruby\Widget.rb"
Size:

**Review by apetrovskiy — CHANGES_REQUESTED** [review]

File: Blender.rb (hunk 1 above)
Comment (artpetrovskii) [reply Blender.rb]: I fixed it

Comment (artpetrovskii) [reply Blender.rb]: I fixed it

File: dog.rb (hunk 4 above)
Comment (artpetrovskii) [reply dog.rb]: I fixed it

File: dog.rb (hunk 5 above)
Comment (artpetrovskii) [reply dog.rb]: I fixed it

Comment (artpetrovskii) [reply dog.rb]: I fixed it

File: animals.rb (hunk 2 above)
Comment (apetrovskiy) [reply animals.rb]: It doesn't work:
```
[Running] ruby "c:\Projects\Ruby\Head-First-Ruby\animals.rb"
c:/Projects/Ruby/Head-First-Ruby/animals.rb:5:in `move': wrong number of arguments (given 1, expected 2) (ArgumentError)
	from c:/Projects/Ruby/Head-First-Ruby/animals.rb:32:in `<main>'

[Done] exited with code=1 in 9.396 seconds
```

File: animals.rb (hunk 3 above)
Comment (artpetrovskii) [reply animals.rb]: I fixed it

File: animals.rb (hunk 2 above)
Comment (artpetrovskii) [reply animals.rb]: I fixed it

File: mage.rb (hunk 6 above)
Comment (artpetrovskii) [reply mage.rb]: I fixed it

File: Blender.rb (hunk 1 above)
Comment (apetrovskiy) [reply Blender.rb]: Resolve conversations, please

File: Blender.rb [hunk 9]
```diff
@@ -0,0 +1,13 @@
+class Blender
```
Comment (apetrovskiy) [inline Blender.rb]: filename

File: dog.rb [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+    if value == ''
+      raise "Name can't be blank!"
+
+    end
+
+@name = value
+  end
+
+  def age=(value)
+    if value < 0
```
Comment (apetrovskiy) [inline dog.rb]: value.negative?

File: MyClass.rb [hunk 11]
```diff
@@ -0,0 +1,15 @@
+class MyClass
```
Comment (apetrovskiy) [inline MyClass.rb]: file name

File: vehicle_methods.rb [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+  puts 'Beep beep!'
+end
+
+def use_headlights(brightness = 'low-beam')
+  puts "Turning on #{brightness} headlights"
+  puts 'Watch out for deer!'
+end
+
+def mileage(miles_driven, gas_used)
+  if gas_used == 0
```
Comment (apetrovskiy) [inline vehicle_methods.rb]: gas_user.zero?

**Review by apetrovskiy — CHANGES_REQUESTED** [review]

File: MyClass.rb (hunk 7 above)
Comment (artpetrovskii) [reply MyClass.rb]: I fixed it

File: Blender.rb (hunk 9 above)
Comment (artpetrovskii) [reply Blender.rb]: I fixed it

File: MyClass.rb (hunk 11 above)
Comment (artpetrovskii) [reply MyClass.rb]: I fixed it

File: dog.rb (hunk 10 above)
Comment (artpetrovskii) [reply dog.rb]: I fixed it

File: vehicle_methods.rb (hunk 12 above)
Comment (artpetrovskii) [reply vehicle_methods.rb]: I fixed it

File: Blender.rb (hunk 9 above)
Comment (artpetrovskii) [reply Blender.rb]: I fixed it

File: MyClass.rb (hunk 11 above)
Comment (artpetrovskii) [reply MyClass.rb]: I fixed it

File: Widget.rb (hunk 8 above)
Comment (artpetrovskii) [reply Widget.rb]: I fixed it

**Review by apetrovskiy — APPROVED** [review]

**Pull request merged**